go/vt/vttablet/endtoend/stream_test.go

```diff
@@ -41,6 +45,71 @@ func TestStreamUnion(t *testing.T) {
 	assert.Equal(t, 1, len(qr.Rows))
 }
 
+func TestStreamLongTail(t *testing.T) {
```

Comment:
Having a hard time following what you are testing here. Mind sprinkling comments to make it easier to follow?